satellite/marketing/marketingweb/server.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package marketingweb
+
+import (
+	"context"
+	"html/template"
+	"net"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+// Error is satellite marketing error type
+var Error = errs.Class("satellite marketing error")
+
+// Config contains configuration for marketing offersweb server
+type Config struct {
+	Address		string `help:"server address of the marketing Admin GUI" default:"0.0.0.0:8090"`
+	StaticDir	string `help:"path to static resources" default:""`
+}
+
+// Server represents marketing offersweb server
+type Server struct {
+	log *zap.Logger
+
+	config Config
+
+	listener net.Listener
+	server   http.Server
+}
+
+// The three pages contained in addPages are pages all templates require
+// This exists in order to limit handler verbosity
+func (s *Server) addPages(assets []string) []string {
+	rp :=  s.config.StaticDir + "/pages/"
+	pages := []string{rp + "base.html", rp + "index.html", rp + "banner.html"}
+	for _, page := range assets {
+		pages = append(pages, page)
+	}
+	return pages
+}
+
+// NewServer creates new instance of offersweb server
+func NewServer(logger *zap.Logger, config Config, listener net.Listener) *Server {
+	server := Server{
+		log:      logger,
+		config:   config,
+		listener: listener,
+	}
+
+	logger.Sugar().Debugf("Starting Marketing Admin UI on %s...", server.listener.Addr().String())
+	fs := http.FileServer(http.Dir(server.config.StaticDir))
+	mux := mux.NewRouter()
+	mux.Handle("/static/", http.StripPrefix("/static",fs))
```

Comment:
Should we check to see if the `StaticDir` is passed in before we pass it into the mux?

Reply:
done